.github/workflows/lint.yml

```diff
@@ -37,60 +75,84 @@ env:
+        if: inputs.language == 'python'
         uses: actions/setup-python@v4
         with:
           python-version: "${{ inputs.python-version }}"
-          cache: "pip"
-          cache-dependency-path: pyproject.toml
+          python-version-file: "${{ inputs.python-version-file }}"
+          cache: "${{ inputs.python-cache-type }}"
+          cache-dependency-path: "${{ inputs.python-cache-dependency-path }}"
 
       # Rust
       - name: configure rust and clippy, if required
-        if: inputs.type == 'rust'
+        if: inputs.language == 'rust' 
+        continue-on-error: ${{ inputs.continue-on-error }}
         run: |
           # we always run the latest stable Rust and Clippy for Rust linting.
           rustup update
           rustup component add clippy
 
       - name: run rustfmt (rust)
-        if: inputs.type == 'rust'
+        if: inputs.language == 'rust' && inputs.type == 'github'
+        continue-on-error: ${{ inputs.continue-on-error }}
         run: cargo fmt --check
 
       - name: run clippy (rust)
         # NOTE: This is a fork of the original `actions-rs/clippy-check`,
         # which is no longer maintained; the commit is pinned since no release
         # has been made and its long term maintenance status/ownership is unclear.
+        if: inputs.language == 'rust' && inputs.type == 'github'
+        continue-on-error: ${{ inputs.continue-on-error }}
         uses: actions-rs-plus/clippy-check@5eb300cdebee2681ff8513b9348b0ca793d8a293
-        if: inputs.type == 'rust'
         with:
           args: --all-features -- -D warnings
 
       - name: run cargo sort (rust)
-        if: inputs.type == 'rust' && inputs.cargo-sort
+        if: inputs.language == 'rust' && inputs.type == 'github'
+        continue-on-error: ${{ inputs.continue-on-error }}
         run: |
           cargo install cargo-sort
           cargo sort -c
 
       # Go
-      - name: setup go
-        if: inputs.type == 'go'
+      - name: configure go, if required
+        if: inputs.language == 'go'
+        continue-on-error: ${{ inputs.continue-on-error }}
         uses: actions/setup-go@v3
         with:
           go-version-file: "${{ inputs.go-version-file }}"
+          go-version: "${{ inputs.go-version }}"
 
       - name: run golangci-lint (go)
-        if: inputs.type == 'go'
+        if: inputs.language == 'go' && inputs.type == 'github'
+        continue-on-error: ${{ inputs.continue-on-error }}
         uses: golangci/golangci-lint-action@0ad9a0988b3973e851ab0a07adf248ec2e100376 # @v3.3.1
         with:
           version: "${{ inputs.golangci-lint-version }}"
 
       # Make
       - name: run lint (make)
         if: inputs.type == 'make'
-        run: make lint
+        continue-on-error: ${{ inputs.continue-on-error }}
+        run: make "${{ inputs.target }}"
+
+      # Just
+      - name: configure just, if required
+        if: inputs.type == 'just'
+        run: cargo install just
```

Comment:
Do you know how long this step takes? I worry about compilation times with Rust on GitHub Actions runners

Reply:
I know I totally agree, it took 1min 45sec last I tried which isn't terrible but isn't great for just a lint. I was using the just github action but hit an API limit. Its possible to add a github PAT to get around that but that's a separate headache, may end up having to do that though

Reply:
> hit an API limit

😞 

Yeah, I've also had issues with downloading from the GitHub Releases page (plus that's brittle when supporting multiple platforms).

Hmmmm. Tough one!

Reply:
Looks like we could cache if we use a different third part action to cargo install: https://github.com/FrancisRussell/ferrous-actions-dev#installing-a-package-with-cargo-install

Reply:
Good call.

Somewhat related: Any idea on the cost calculation for caching in private repos, if these templates are also to be used there?

Reply:
FWIW, `just` also provides builds: https://github.com/casey/just/releases

(I'm not saying we *should* use those, but it would shortcut the build here.)

Reply:
@ekilmer mind creating a follow-up issue for this? I'm going to get this merged today so we can experiment with it further 🙂

Reply:
Opened https://github.com/trailofbits/.github/issues/11